src/renderer/canvas/canvas-renderer.ts

```diff
@@ -245,6 +245,47 @@ export class CanvasRenderer<N extends INodeBase, E extends IEdgeBase> extends Em
     return zoomIdentity.translate(newX, newY).scale(newZoom);
   }
 
+  getZoomTransform(zoomFactor: number): ZoomTransform {
+    const previousZoom = this.transform.k;
+    const newZoom = Math.max(
+      Math.min(zoomFactor * previousZoom, this._settings.maxZoom),
+      this._settings.minZoom
+    );
+    return zoomIdentity.scale(newZoom);
+  }
+
+  getDragLeftTransform(draggingFactor: number): ZoomTransform {
+    const currentTransform = this.transform;
+    const newX = currentTransform.x + draggingFactor;
+    return zoomIdentity
+      .translate(newX, currentTransform.y)
+      .scale(currentTransform.k);
+  }
+
+  getDragRightTransform(draggingFactor: number): ZoomTransform {
+    const currentTransform = this.transform;
+    const newX = currentTransform.x - draggingFactor;
+    return zoomIdentity
+      .translate(newX, currentTransform.y)
+      .scale(currentTransform.k);
+  }
+
+  getDragUpTransform(draggingFactor: number): ZoomTransform {
+    const currentTransform = this.transform;
+    const newY = currentTransform.y + draggingFactor;
+    return zoomIdentity
+      .translate(currentTransform.x, newY)
+      .scale(currentTransform.k);
+  }
+
+  getDragDownTransform(draggingFactor: number): ZoomTransform {
+    const currentTransform = this.transform;
+    const newY = currentTransform.y - draggingFactor;
+    return zoomIdentity
+      .translate(currentTransform.x, newY)
+      .scale(currentTransform.k);
+  }
```

Comment:
All these are pretty much doing the same thing, so I would propose having one function that handles all 4 cases with enum, e.g. `PanDirectionType.{UP, DOWN, LEFT, RIGHT}`.

Btw this event should be `pan`, `drag` is used for dragging nodes/edges. Pan is when you move the canvas up/down/left/right.
With the enum, you can have just one function `getPanTransform(panDirectionType, factor);`

Reply:
Created single `getPanTransform` function, Pls check.